src/vm-repair/azext_vm_repair/custom.py

```diff
@@ -644,7 +646,20 @@ def reset_nic(cmd, vm_name, resource_group_name, yes=False):
             update_ip_command = 'az network nic ip-config update -g {g} --nic-name {nic} -n {config} --private-ip-address {ip}' \
                                 .format(g=resource_group_name, nic=primary_nic_name, config=ipconfig_name, ip=swap_ip_address)
         _call_az_command(update_ip_command)
-
+            # Check if IP is updated
+    while True:
+        get_current_ip_command = 'az network nic ip-config wait --updatede -g {g} --nic-name {nic} -n ' \
+                                .format(g=resource_group_name, nic=primary_nic_name)
+        
+        current_ip_address = _call_az_command(get_current_ip_command)
+        if current_ip_address == swap_ip_address:
+            logger.info('IP address updated successfully to {ip}\n'.format(ip=swap_ip_address))
+            break
+        else:
+            logger.info('IP address not updated yet. Waiting for a few minutes before retrying...\n')
+            time.sleep(300)
+            # Sleep for 5 minutes 
```

Comment:
Are you sure the customer needs to wait for such a long time? I'm not sure if the customer has the patience to wait for 5 minutes